axelrod/match_generator.py

```diff
@@ -167,3 +167,61 @@ def estimated_size(self):
         """Rough estimate of the number of matches that will be generated."""
         size = self.__len__() * (1. / self.prob_end) * self.repetitions
         return size
+
+class SpatialMatches(RoundRobinMatches):
+
+    def __init__(self, players, turns, game, repetitions, edges):
+
+        player_indices = list(range(len(players)))
+        node_indices = sorted(set([node for edge in edges for node in edge]))
+        if player_indices != node_indices:
+            raise ValueError("The graph edges do not include all players.")
+
+        self.edges = edges
+        super(SpatialMatches, self).__init__(players, turns, game, repetitions)
+        """
+        A class that generates spatial structure matches. In spatial structure
```

Comment:
A class that generates *spatially-structured matches.* In these matches, players interact only with their neighbors rather than the entire population. This reduces to a well-mixed population when the spatial graph is a complete graph.